frontend-react/src/components/Snitch.tsx

```diff
@@ -94,46 +137,62 @@ interface PayloadProps {
 }
 function Payload(props: PayloadProps) {
     const { title, message } = props;
+    let contentBadges;
+    if (message.contentLength && message.contentLength > 0) {
+        contentBadges = <div className="float-right">
+            <span className="badge badge-primary">{message.contentType}</span>
+            &nbsp;
+            <span className={`badge badge-primary`}>{message.contentLength}</span>
+        </div>
+    }
+
     return (
         <Card>
-            <div className="card-header">
+            <Card.Header>
                 {title}
-            </div>
-            <Card.Body>
+                {contentBadges}
+            </Card.Header>
+            <Card.Body className="pb-0">
                 <dl className="row">
                     {Object.entries(message.headers).map(([key, value]) =>
-                        <>
-                            <dt className="col-sm-3">{key}</dt>
-                            <dd className="col-sm-9">{value}</dd>
-                        </>
+                        <React.Fragment key={key}>
+                            <dt className="col-sm-3 col-md-2">{key}</dt>
+                            <dd className="col-sm-9 col-md-10">{value}</dd>
+                        </React.Fragment>
                     )}
                 </dl>
             </Card.Body>
-            <Card.Body>
-                <PayloadBody payload={message} />
-            </Card.Body>
+            <PayloadBody payload={message} />
         </Card>
     );
 }
 
+/**
+ * Creates a unique sorted array of the given RequestResponse array.
+ */
+function uniqueAndSorted(requestResponses: RequestResponse[]) {
+    return Array.from(new Set(requestResponses)).sort((r1, r2) => r1.timestamp > r2.timestamp ? -1 : 1);
```

Comment:
Er du sikker på at det med Set frem og tilbake faktisk gjør ønsket oppførsel? To `requestResponse`-objekter kan jo være strukturelt like, men fortsatt ulike objekter?

Reply:
De inneholder et id-felt som gjør de unike.